flash/video/classification/data.py

```diff
@@ -231,6 +535,137 @@ def from_csv(
+            test_resolver: Optionally provide a function which converts an entry from the ``input_field`` into a video
+                file path.
+            predict_file: The CSV file to use when predicting.
+            predict_videos_root: The root directory containing predict videos.
+            predict_resolver: Optionally provide a function which converts an entry from the ``input_field`` into a
+                video file path.
+            train_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when training.
+            val_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when validating.
+            test_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when testing.
+            predict_transform: The :class:`~flash.core.data.io.input_transform.InputTransform` type to use when
+                predicting.
+            clip_sampler: The clip sampler to use. One of: ``"uniform"``, ``"random"``, ``"constant_clips_per_video"``.
+            clip_duration: The duration of clips to sample.
+            clip_sampler_kwargs: Additional keyword arguments to use when constructing the clip sampler.
+            video_sampler: Sampler for the internal video container. This defines the order videos are decoded and,
+                if necessary, the distributed split.
+            decode_audio: If True, also decode audio from video.
+            decoder: The decoder to use to decode videos. One of: ``"pyav"``, ``"torchvision"``. Not used for frame
+                videos.
+            input_cls: The :class:`~flash.core.data.io.input.Input` type to use for loading the data.
+            predict_input_cls: The :class:`~flash.core.data.io.input.Input` type to use for loading the prediction data.
+            transform_kwargs: Dict of keyword arguments to be provided when instantiating the transforms.
+            data_module_kwargs: Additional keyword arguments to provide to the
+                :class:`~flash.core.data.data_module.DataModule` constructor.
+
+        Returns:
+            The constructed :class:`~flash.video.classification.data.VideoClassificationData`.
+
+        Examples
+        ________
+
+        .. testsetup::
+
+            >>> import os
+            >>> import torch
+            >>> from torchvision import io
+            >>> from pandas import DataFrame
+            >>> data = torch.randint(255, (10, 64, 64, 3))
+            >>> os.makedirs("train_folder", exist_ok=True)
+            >>> os.makedirs("predict_folder", exist_ok=True)
+            >>> io.write_video(os.path.join("train_folder", "video_1.mp4"), data, 5, "libx264rgb", {"crf": "0"})
+            >>> io.write_video(os.path.join("train_folder", "video_2.mp4"), data, 5, "libx264rgb", {"crf": "0"})
+            >>> io.write_video(os.path.join("train_folder", "video_3.mp4"), data, 5, "libx264rgb", {"crf": "0"})
```

Comment:
```suggestion
            >>> _ = [io.write_video(os.path.join("train_folder", f"video_{i}.mp4"), data, 5, "libx264rgb", {"crf": "0"}) for i in range(1, 4)]
```
just wondering if there can be `testsetup` common for all modules as for example, you create the same files for each test and in fact you are fine to create them just for module start and clean at the end

Reply:
Eventually these will be moved to `tests/helpers` so you get a good balance of readability and simplicity, but this will be done in a follow-up when we overhaul the tests :smiley: